File: old/Marcianitos/sketch.js
let spaceShip;
let start = false;
let lasers = [];
let penguins = [];
let stars;

function preload()  {
    shipImg = loadImage('../files/ship.png');
    penguinImg = loadImage('../files/spacePenguin.png');
    heartImg = loadImage('../files/heart.png');
    
}

function setup()    {
    createCanvas(600, 690);
    spaceShip = new Ship();
    stars = new Stars();
}

function draw()    {
    background(0);
    
    stars.update();
    
    spaceShip.update();
    spaceShip.render();
    
    for (let i=0; i<lasers.length; i++) {
        lasers[i].update();
        lasers[i].render();
    
        let hit = lasers[i].hits(penguins);
        if (hit)    {
            penguins.splice(hit - 1, 1);
            lasers.splice(i, 1);
        } else if (lasers[i].offScreen())   {
            lasers.splice(i, 1);
    
        }
    }
    
    for (let i=0; i<penguins.length; i++)   {
        penguins[i].update();
        penguins[i].render();
    }
    
    
    game();
}

function keyPressed() {
    switch (key)    {
        case ' ':
            lasers.push(new Laser(spaceShip.posX, spaceShip.posY - spaceShip.size/2));
            break;
        case 's':
            start = true;
            loop();
            break;
        case 'p':
            start = false;
            noLoop();
            break;
        default:
            break;
    }
}

function game() {
    if (!start) {
        noLoop();
        textSize(32);
        fill('white');
        textAlign(CENTER);
        text('PRESS S TO START', width/2, height/2);
    } else  {
        if (random(0, 100) < 1)    {
            penguins.push(new Penguin());
        }
    }
}

File: sketch.py
from p5 import *
from Spaceship import Spaceship
from Stars import Stars
from Enemy import Enemy
from Laser import Laser
from random import randint

height = 800
width = 600
start = False
gameOver = False
lasers = []
enemies = []
spaceship = None
stars = None
frame = 0
time = 0

shipImg = None
startImg = None
spaceInvImg = None
gameOverImg = None
explosionImages = []

def imagesPreload():
  global startImg
  startImg = load_image('./assets/start-text.png')

  global spaceInvImg
  spaceInvImg = load_image('./assets/Title.png')

  global gameOverImg
  gameOverImg = load_image('./assets/GameOver.png')

  global explosionImg
  for i in range(12):
    explosionImages.append(load_image('./assets/explosion/' + str(i) + '.png'))

def setup():
  # Creates window
  size(width, height)
  # Sets all drawing as white
  stroke(255)
  #no_loop()
  global spaceship
  spaceship = Spaceship(width, height)

  global stars
  stars = Stars(width, height)

  imagesPreload()

  
def draw():
  global gameOver
  # Sets black background
  background(0)

  if not gameOver:
    spaceship.update()
  spaceship.render()
  stars.update()

  for laser in lasers:
    laser.update();
    laser.render();

    hit = laser.hits(enemies);
    if hit:
      del enemies[hit - 1]
      del lasers[lasers.index(laser)]
    elif laser.offScreen():  
      del lasers[lasers.index(laser)]

  for enemy in enemies:
    enemy.update()
    enemy.render()

    if spaceship.hits(enemy) or enemy.offScreen():
      gameOver = True


  game()


def key_pressed():
  global start

  if key == ' ':
    if not gameOver:
      lasers.append(Laser(spaceship.posX, spaceship.posY - spaceship.size/2, height))
  elif (key == 's' or key == 'S') and not gameOver:
    start = True
    loop()
  elif key == 'p' or key == 'P':
    start = False
    no_loop()

def game():
  if not start:
    no_loop()
    image(spaceInvImg, (width/2, height/3))
    image(startImg, (width/2, height/2))
    
  elif gameOver:
    image(gameOverImg, (width/2, height/2))

    global frame
    frame = (frame + 1) % 12
    image(
      explosionImages[frame],
      (
        spaceship.posX + randint(-spaceship.size/2, spaceship.size/2),
        spaceship.posY + randint(-spaceship.size/2, spaceship.size/2)
      )
    )

  elif randint(1, 100) <= 3:
    enemies.append(Enemy(width, height))

if __name__ == '__main__':
    run()